packages/js/data/src/reviews/index.ts

```diff
@@ -12,14 +14,19 @@ import * as selectors from './selectors';
 import * as actions from './actions';
 import * as resolvers from './resolvers';
 import controls from '../controls';
-import reducer from './reducer';
+import reducer, { State } from './reducer';
 
-registerStore( STORE_NAME, {
-	reducer,
+export * from './types';
+export type { State };
+
+registerStore< State >( STORE_NAME, {
+	reducer: reducer as Reducer< State, AnyAction >,
 	actions,
 	controls,
 	selectors,
 	resolvers,
 } );
 
 export const REVIEWS_STORE_NAME = STORE_NAME;
+
```

Comment:
Hey @rjchow I suggest that we can add a declaration like below so that we can get the right types automatically without setting the selector type.

https://github.com/woocommerce/woocommerce/blob/b8066f9b3115cea640e259c34136141633a6a6cd/packages/js/data/src/notes/index.ts#L32-L39

Reply:
Will do, thanks!